common/types/shared_test.go

```diff
@@ -0,0 +1,317 @@
+	expectErr := error(nil)
+
+	assert.Equal(t, expectRes, serializeRes)
+	assert.Equal(t, expectErr, err)
+
+	// test serializing a request that only contains PII
+	testPiiReq := createPiiSampleRequest()
+	serializePiiRes, piiErr := SerializeRequest(testPiiReq)
+
+	expectPiiRes := "{}"
+	expectPiiErr := error(nil)
+
+	assert.Equal(t, expectPiiRes, serializePiiRes)
+	assert.Equal(t, expectPiiErr, piiErr)
+
+	assert.NotPanics(t, func() {
+		SerializeRequest(nil)
+	})
+
+}
+
+func createNewSignalWithStartWorkflowExecutionRequest() *SignalWithStartWorkflowExecutionRequest {
+	testTasklistKind := TaskListKind(1)
+	testExecutionStartToCloseTimeoutSeconds := int32(1)
+	testTaskStartToCloseTimeoutSeconds := int32(1)
+	testWorkflowIDReusePolicy := WorkflowIDReusePolicy(1)
+	testDelayStartSeconds := int32(1)
+	testJitterStartSeconds := int32(1)
+	testReq := &SignalWithStartWorkflowExecutionRequest{
+		Domain:       "testDomain",
+		WorkflowID:   "testWorkflowID",
+		WorkflowType: &WorkflowType{Name: "testWorkflowType"},
+		TaskList: &TaskList{
+			Name: "testTaskList",
+			Kind: &testTasklistKind,
+		},
+		Input:                               []byte("testInputPII"),
+		ExecutionStartToCloseTimeoutSeconds: &testExecutionStartToCloseTimeoutSeconds,
+		TaskStartToCloseTimeoutSeconds:      &testTaskStartToCloseTimeoutSeconds,
+		Identity:                            "testIdentity",
+		RequestID:                           "DF66E35D-A5B0-425D-8731-6AAC4A4B6368",
+		WorkflowIDReusePolicy:               &testWorkflowIDReusePolicy,
+		SignalName:                          "testRequest",
+		SignalInput:                         []byte("testSignalInputPII"),
+		Control:                             []byte("testControl"),
+		RetryPolicy: &RetryPolicy{
+			InitialIntervalInSeconds:    1,
+			BackoffCoefficient:          1,
+			MaximumIntervalInSeconds:    1,
+			MaximumAttempts:             1,
+			NonRetriableErrorReasons:    []string{"testArray"},
+			ExpirationIntervalInSeconds: 1,
+		},
+		CronSchedule:       "testSchedule",
+		Memo:               &Memo{Fields: map[string][]byte{}},
+		SearchAttributes:   &SearchAttributes{IndexedFields: map[string][]byte{}},
+		Header:             &Header{Fields: map[string][]byte{}},
+		DelayStartSeconds:  &testDelayStartSeconds,
+		JitterStartSeconds: &testJitterStartSeconds,
+	}
+	return testReq
+}
+
+type PiiSampleRequest struct {
+	Input            []byte            `json:"-"` // Filtering PII
```

Comment:
this is effectively testing `encoding/json`, which we can broadly assume is correct.

what we want is to ensure we don't start logging these fields _on the real objects_.  this won't protect against that at all.

that said: creating a pii-field-only instance of the struct and comparing against `{}` seems effective!  I think that'll work on the real things too, since everything(?) is `omitempty`.

Reply:
deleted PiiSampleRequest